### ISSUE TITLE
Documentation improvements for Dataset

### ISSUE THREAD

**Issue by kevinrsoursib** [andrewdalpino/Rubix-engine#348]
The documentation for the Dataset class is ambiguous and confusing.  In particular it's hard to follow which functions return a reference to the current Dataset (like transform) after modification in place and which (like filter) make return a newly created dataset copying some or all of the current dataset.  In particular the way the filter function is described implies modification in place which is not what it does.

**Comment by andrewdalpino** [comment]
Hi @kevinrsoursib thanks for the feedback. Would you like to help clarify the documentation? I agree that the documentation could be clearer but because the project is so underfunded (it's basically just me funding it with my own time and money) we cannot actually make improvements without the help of the community such as yourself.

Thanks and feel free to reach out if I can help get you started.
